logging/logger_test.go

```diff
@@ -149,3 +151,62 @@ func TestLogger_SetupLoggerWithJSON(t *testing.T) {
 	require.Contains(jsonOutput, "@message")
 	require.Equal(jsonOutput["@message"], "test warn msg")
 }
+
+func TestLogger_SetupLoggerWithValidLogPath(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+
+	tmpDir := testutil.TempDir(t, t.Name())
+	defer os.RemoveAll(tmpDir)
+
+	cfg := &Config{
+		LogLevel:    "INFO",
+		LogFilePath: tmpDir + "/",
+	}
+	ui := cli.NewMockUi()
+
+	logger, gatedWriter, writer, ok := Setup(cfg, ui)
+	require.True(ok)
+	require.NotNil(logger)
+	require.NotNil(gatedWriter)
+	require.NotNil(writer)
+}
+
+func TestLogger_SetupLoggerWithInValidLogPath(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+
+	cfg := &Config{
+		LogLevel:    "INFO",
+		LogFilePath: "nonexistentdir/",
+	}
+	ui := cli.NewMockUi()
+
+	logger, gatedWriter, writer, ok := Setup(cfg, ui)
+	require.Contains(ui.ErrorWriter.String(), "no such file or directory")
+	require.False(ok)
+	require.Nil(logger)
+	require.Nil(gatedWriter)
+	require.Nil(writer)
+}
+
+func TestLogger_SetupLoggerWithInValidLogPathPermission(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+
+	os.Mkdir("logs", 0000)
+	defer os.RemoveAll("./logs")
+
```

Comment:
I could not find a way to create tmpdir with custom permissions.
Also I founf ioutil.TempDir does not allow setting custom permission.